e2e/path/main.go

```diff
@@ -74,6 +74,22 @@ var tests = []test{
 		},
 		expected: "https://fallback-to.path.path.example.com",
 	},
+	{
+		name: "Redirect using the use= field and upstream record",
+		args: data{
+			host: "sourcerecord.path.path.example.com",
+			path: "/testing",
+		},
+		expected: "https://upstream.path.path.example.com",
+	},
+	{
+		name: "Redirect to upstream record using multiple use= fields",
+		args: data{
+			host: "srcrecord.path.path.example.com",
+			path: "/testing",
+		},
+		expected: "https://upstream.path.path.example.com",
+	},
```

Comment:
Adding an additional test, that uses `use=`, but that does go into fallback. Aka there is no valid upstream working.
That seems to be a good additional failure case.